BUG: Auto credits PR bot creates zombie branch when fails

Example log: https://github.com/astropy/astropy/actions/runs/29723674437/job/88291853803

`Run gh pr create`

`pull request create failed: GraphQL: No commits between main and update-credits-1784531341 (createPullRequest)`

`Error: Process completed with exit code 1.`

As a result, a branch called `update-credits-1784531341` was created in this repo but a pull request was never opened from it. (I had since manually deleted it but the general problem remains.)

Desired outcome: When there is no commit (which is possible when we don't gain any new contributor during the time period), quit the workflow gracefully before creating a branch. If that is not possible, then quit the workflow gracefully and delete the branch. I do not like the second option because it gives the bot dangerous permission to delete branches. If branch must be created/deleted after the fact, maybe we should move this workflow to a special bot fork (like what the backport bot does) instead.

What do you think, @astrofrog and @neutrinoceros ?

--- a/.github/workflows/update_credits.yml
+++ b/.github/workflows/update_credits.yml
@@ -44,6 +44,7 @@ jobs:
     - name: Show diff
       run: git diff docs/credits.rst
     - name: Commit changes
+      id: commit
       if: github.event_name != 'pull_request'
       run: |
         git config user.name github-actions
@@ -53,11 +54,14 @@ jobs:
         git add docs/credits.rst
         if ! git diff --cached --exit-code; then
           git commit -m "Update docs/credits.rst with new contributors"
+          echo "pr-needed=true" >> "$GITHUB_OUTPUT"
+        else
+          echo "pr-needed=false" >> "$GITHUB_OUTPUT"
         fi
-        git push --set-upstream origin "$BRANCH_NAME"
     - name: Create Pull Request
-      if: github.event_name != 'pull_request'
+      if: ${{ steps.commit.outputs.pr-needed && github.event_name != 'pull_request' }}
       run: |
+        git push --set-upstream origin "$BRANCH_NAME"
         gh pr create \
           --title "Update credits to add new contributors" \
           --label no-changelog-entry-needed --label Docs \

